packages/ia-userlist-settings/src/user-lists-service/user-lists-service-interface.ts

```diff
@@ -0,0 +1,112 @@
+import type { Result } from '@internetarchive/result-type';
+import type { UserListsError } from './user-lists-error';
+import type { SearchResult } from '@internetarchive/search-service';
+import type {
+  UserList,
+  UserListMember,
+  UserListMemberOptions,
+  UserListOptions,
+} from './models';
+
+export interface UserListsServiceInterface {
+  /**
+   * Fetches the full set of the given user's lists from the backend service.
+   * The result does not include the members array for each list; only the list metadata.
+   *
+   * If fetching lists for the currently logged-in user, their user id will automatically
+   * be replaced by `me` in the endpoint URL.
+   *
+   * @param userId The ID of the user to fetch lists for (e.g., `@brewster`)
+   */
+  fetchAllListsForUser(
+    userId: string,
+  ): Promise<Result<UserList[], UserListsError>>;
+
+  /**
+   * Fetches a specific user list, including the list metadata and the array of list
+   * members. Note that the list members returned include only the basic information
+   * kept in the user lists store, not the full item metadata. For that, use `fetchListMembers`.
+   *
+   * @param userId The ID of the user whose list should be fetched
+   * @param listId The ID of the list to fetch (with respect to the given user)
+   */
+  fetchList(
```

Comment:
MINOR SUGGESTION: Consider `fetchListForUser` as the name to parallel `fetchAllListsForUser` and cuz in both cases, the first parameter is the `userId` so the function name reads well going into its first parameter.

Reply:
OTOH: Maybe you want the API to be as "list" focused as possible. (Cuz maybe the same argument applies to `fetchListMembers`.

Reply:
I would propose the following names for the functions.
`FetchLists()` -> Fetch all lists
`FetchList()` -> Fetch single list
`FetchListMembers()` 
`FetchListsForItem()` -> fetch lists for details page to shows if item is already in xxx list

Reply:
I suppose the question is whether the `ForUser` bit adds anything useful.

My original thinking around this was to avoid having two different method names differing by a single `s`, and just be explicitly clear about which one is user-encompassing instead. The code will be read far more often than written after all, and relying on a one-letter difference to signify a large difference in functionality sets off alarm bells for me.

But on the other hand, I do agree that it feels slightly redundant given that these are already explicitly _user_ lists.

Reply:
Had some in-person chat about this, and I think we settled on the following:

- `fetchListsForUser(userId)` -- returns all lists owned by the given user
- `fetchList(userId, listId)` -- returns a single, specific list, as identified by the `userId`/`listId` pair
- `fetchListMembers(userId, listId)` -- returns the array of members for a specific list
- `fetchOwnListsContainingItem(itemId)` -- returns all of the _current_ user's lists which contain the given item identifier

The rationale is a mixture of clarity and future-proofing. We are likely to eventually extend this service with methods to fetch lists across _multiple_ users. If we reserve a very general method name like `fetchLists` for the _user-specific_ method now, it could hamper us in the future to be stuck with that among _even more general_ cases that have to tiptoe around the existing name. Better to be explicit & clear about what the method does.

Likewise, `fetchOwnListsContainingItem` makes it clear upfront that the method acts on the _current user's_ lists, and allows us to cleanly extend the service with methods that act on a more general set of users. The name `fetchListsForItem` doesn't fully convey these semantics.